### PR TITLE
[2656] - refactor: Registration verification from XML to compose

### LINKED ISSUE #2584: refactor: migration of RegistrationVerificationFragment to compose screen
**Summary:**

migration of RegistrationVerificationFragment to compose screen

### PR DESCRIPTION
**Pull request by akashmeruva9**
Fixes #2584 

<table style="border: 1px solid black;">
            <tr>
                <td  style="border: 1px solid black ;">
                    
                </td>
                <td  style="border: 1px solid black ;">
                    
                </td>
                <td  style="border: 1px solid black ;">
                    
                </td>
            </tr>
</table>

<table style="border: 1px solid black;">
            <tr>
                <td  style="border: 1px solid black ;">
                    
</table>

https://github.com/openMF/mifos-mobile/assets/106223361/a603b5da-58f8-4217-93e0-32c776a0f2ad

- [ x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x ] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x ] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationFragment.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+@AndroidEntryPoint
+class RegistrationVerificationFragment : BaseFragment() {
+
+    private val registrationViewModel: RegistrationViewModel by viewModels()
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        return mifosComposeView(requireContext()) {
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationFragment.kt:27]: set a view composition strategy below this line

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationFragment_Old.kt [hunk 2]
```diff
@@ -18,13 +18,12 @@ import org.mifos.mobile.ui.login.LoginActivity
...
 import org.mifos.mobile.utils.Toaster
-import org.mifos.mobile.ui.registration.RegistrationViewModel
 
 /**
  * Created by dilpreet on 31/7/17.
  */
 @AndroidEntryPoint
-class RegistrationVerificationFragment : BaseFragment() {
+class RegistrationVerificationFragment_Old : BaseFragment() {
     private var _binding: FragmentRegistrationVerificationBinding? = null
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationFragment_Old.kt:27]: remove this fragment if not needed

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationFragment.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            RegistrationVerificationScreen(
+                registrationViewModel,
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationFragment.kt]: no need of passing viewModel. we can omit this parameter as we already assigned hiltViewModel

**Review by PratyushSingh07 — CHANGES_REQUESTED** [review]
i see that you have attached some screemshots and I was wondering if they are the same as the earlier XML designs. Also in the error state I think you should remove the Mifos Icon , it looks a bit odd . What do you have to say about this @AvneetSingh2001 ?

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    navigateBack: () -> Unit,
+    finishActivity: ()->Unit)
+{
+    val context = LocalContext.current
+    val uiState by viewModel.registrationVerificationUiState.collectAsStateWithLifecycle()
+
+    RegistrationVerificationContent(uiState = uiState,
+        verifyUser = { token, id -> viewModel.verifyUser(token, id) },
+        retryConnection = { token, id -> viewModel.verifyUser(token, id) },
+        finishActivity = finishActivity)
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt]: please do indentation. 
```
RegistrationVerificationContent(
        uiState = uiState,
        verifyUser = { token, id -> viewModel.verifyUser(token, id) },
        retryConnection = { token, id -> viewModel.verifyUser(token, id) },
        finishActivity = finishActivity
)
        
```

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationFragment_Old.kt (hunk 2 above)
Comment (akashmeruva9) [reply app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationFragment_Old.kt:27]: Rajan sir asked to add the old fragments in the meeting, that's why i am keeping them

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+}
+
+
+@Composable
+    fun RegistrationVerificationContent(
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt]: Use MFScaffold, and pass this content in the scaffoldContent of the scaffold

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+                            verifyUser(authenticationToken.toString(), requestID.toString())
+                        },
+                        Modifier
+                            .fillMaxWidth()
+                            .padding(start = 16.dp, end = 16.dp, top = 4.dp),
+                        contentPadding = PaddingValues(12.dp),
+                        colors = ButtonDefaults.buttonColors(
+                            containerColor = if (isSystemInDarkTheme()) Color(
+                                0xFF9bb1e3
+                            ) else Color(0xFF325ca8)
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt]: single line if

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+                MifosProgressIndicator(
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .padding(10.dp)
+                        .background(MaterialTheme.colorScheme.background.copy(alpha = 0.7f))
+                )
+            }
+
+             RegistrationUiState.Success -> {
+                context.startActivity(Intent(context, LoginActivity::class.java))
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt]: dont start activity from here, do this in fragment, the callback should call this LoginActivity from fragment

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+                        keyboardType = KeyboardType.Number
+                    )
+
+                    MifosOutlinedTextField(
+                        value = authenticationToken,
+                        onValueChange = { authenticationToken = it },
+                        label = R.string.authentication_token,
+                        supportingText = "",
+                        keyboardType = KeyboardType.Number
+                    )
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt]: do error handling as well for these textfields

**Review by AvneetSingh2001 — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationFragment.kt (hunk 1 above)
Comment (AvneetSingh2001) [reply app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationFragment.kt:27]: view Composition strategy is implemented in mifosComposeView already, so no need

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+                    RegistrationVerificationContent(verifyUser)
+                }
+
+                RegistrationUiState.Loading -> {
+                    MifosProgressIndicator(
+                        modifier = Modifier
+                            .fillMaxSize()
+                            .padding(10.dp)
+                            .background(MaterialTheme.colorScheme.background.copy(alpha = 0.7f))
+                    )
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt]: User MifosProgressOverlayIndicator

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+            Image(
+                painterResource(R.drawable.mifos_logo),
+                contentDescription = "",
+                contentScale = ContentScale.Fit,
+                alignment = Alignment.Center,
+                modifier = Modifier
+                    .padding(dimensionResource(id = R.dimen.Mifos_DesignSystem_Spacing_screenHorizontalMargin))
+                    .height(dimensionResource(id = R.dimen.Mifos_DesignSystem_Size_LogoImageSize))
+                    .fillMaxWidth()
+            )
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt]: move this in content screen, show image only in the case of Initial state

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt [hunk 11]
```diff
@@ -0,0 +1,269 @@
...
+        )
+
+        Button(
+            onClick = {
+                if(validateInput())
+                    verifyUser(authenticationToken.toString(), requestID.toString())
+            },
+            Modifier
+                .fillMaxWidth()
+                .padding(start = 16.dp, end = 16.dp, top = 4.dp),
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt]: give top padding 12

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt [hunk 12]
```diff
@@ -0,0 +1,269 @@
...
+
+    if (showConfirmationDialog) {
+        AlertDialog(
+            onDismissRequest = { showConfirmationDialog = false },
+            title = { Text(text = getString(context, R.string.dialog_cancel_registration_title)) },
+            text = { Text(text = getString(context, R.string.dialog_cancel_registration_message)) },
+            confirmButton = {
+                TextButton(onClick = { showConfirmationDialog = false
+                    navigateBack.invoke() }) {
+                    Text(text = "Yes")
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt]: use stringResource

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt [hunk 13]
```diff
@@ -0,0 +1,269 @@
...
+                }
+            },
+            dismissButton = {
+                TextButton(onClick = { showConfirmationDialog = false }) {
+                    Text(text = "No")
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt]: use stringResource

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt [hunk 14]
```diff
@@ -0,0 +1,269 @@
...
+        navigateBack = navigateBack  ,
+        scaffoldContent = { contentPadding->
+
+        Column(modifier = Modifier
+            .padding(contentPadding)
+            .fillMaxSize()) {
+
+            Image(
+                painterResource(R.drawable.mifos_logo),
+                contentDescription = "",
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt]: pass null as contentDesc

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt [hunk 15]
```diff
@@ -0,0 +1,269 @@
...
+        topBarTitleResId = R.string.register,
+        navigateBack = navigateBack  ,
+        scaffoldContent = { contentPadding->
+
+        Column(modifier = Modifier
+            .padding(contentPadding)
+            .fillMaxSize()) {
+
+            Image(
+                painterResource(R.drawable.mifos_logo),
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt]: painter = painterResource(R.drawable.mifos_logo),

**Review by AvneetSingh2001 — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationActivity.kt [hunk 16]
```diff
@@ -19,13 +19,7 @@ class RegistrationActivity : BaseActivity() {
...
     override fun onBackPressed() {
-        MaterialDialog.Builder().init(this)
-            .setTitle(getString(R.string.dialog_cancel_registration_title))
-            .setMessage(getString(R.string.dialog_cancel_registration_message))
-            .setPositiveButton(getString(R.string.yes)) { _, _ -> super.onBackPressed() }
-            .setNegativeButton(R.string.no) { dialog, _ -> dialog.dismiss() }
-            .createMaterialDialog()
-            .show()
+       super.onBackPressed()
     }
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationActivity.kt]: remove this function

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt [hunk 17]
```diff
@@ -0,0 +1,276 @@
...
+fun RegistrationVerificationScreen(
+    navigateBack: () -> Unit?,
+    onVerified: () -> Unit
+) {
+
+    val context = LocalContext.current
+    var showConfirmationDialog by remember { mutableStateOf(false) }
+
+    BackHandler(enabled = true)
+    {
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt]: BackHandler(endabled = true) {

do proper indentation

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt [hunk 18]
```diff
@@ -0,0 +1,276 @@
...
+        mutableStateOf(TextFieldValue(""))
+    }
+    var requestIDError by remember { mutableStateOf(false) }
+    var authenticationTokenError by remember { mutableStateOf(false) }
+
+    fun validateInput(): Boolean {
+
+        var temp = true
+        if (requestID.text.isEmpty()) {
+            requestIDError = true;
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt]: remove semicolon

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt [hunk 19]
```diff
@@ -0,0 +1,276 @@
...
+        )
+
+        MifosOutlinedTextField(
+            value = requestID,
+            onValueChange = {
+                requestID = it
+                requestIDError = false
+            },
+            label = R.string.request_id,
+            supportingText = "Request ID cannot be empty",
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt]: use stringResource

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt [hunk 20]
```diff
@@ -0,0 +1,276 @@
...
+        )
+
+        MifosOutlinedTextField(
+            value = authenticationToken,
+            onValueChange = {
+                authenticationToken = it
+                authenticationTokenError = false
+            },
+            label = R.string.authentication_token,
+            supportingText = "Authentication Token cannot be empty",
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt]: use stringResource

**Review by AvneetSingh2001 — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt [hunk 21]
```diff
@@ -0,0 +1,277 @@
...
+        )
+
+        Button(
+            onClick = {
+                if (validateInput())
+                    verifyUser(authenticationToken.toString(), requestID.toString())
+            },
+            Modifier
+                .fillMaxWidth()
+                .padding(start = 12.dp, end = 16.dp, top = 4.dp),
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt]: not start padding, top padding should be 12, keep start 16

**Review by AvneetSingh2001 — CHANGES_REQUESTED** [review]

**Pull request merged**